build/lint.mk

```diff
@@ -0,0 +1,38 @@
+################################################################################
+###                             Required Variables                           ###
+################################################################################
+ifndef DOCKER
+$(error DOCKER not set)
+endif
+
+ifndef BUILD_CACHE_DIR
+$(error BUILD_CACHE_DIR not set)
+endif
+
+################################################################################
+###                             Lint Settings                                ###
+################################################################################
+
+LINT_FROM_REV ?= $(shell git merge-base origin/master HEAD)
+
+GOLANGCI_VERSION ?= $(shell cat .golangci-version)
+GOLANGCI_IMAGE_TAG ?= golangci/golangci-lint:$(GOLANGCI_VERSION)
+
+GOLANGCI_CACHE_DIR ?= $(CURDIR)/$(BUILD_CACHE_DIR)/golangci-lint/$(GOLANGCI_VERSION)
```

Comment:
`make clean` removes the build cache directory, which is nice when the lint cache goes astray